Sources/Sentry/SentryAppStartTracker.m

```diff
@@ -0,0 +1,172 @@
+#import "SentryAppStartMeasurement.h"
+#import "SentryAppStateManager.h"
+#import "SentryLog.h"
+#import <Foundation/Foundation.h>
+#import <SentryAppStartTracker.h>
+#import <SentryAppState.h>
+#import <SentryClient+Private.h>
+#import <SentryCurrentDateProvider.h>
+#import <SentryDispatchQueueWrapper.h>
+#import <SentryFileManager.h>
+#import <SentryHub.h>
+#import <SentryInternalNotificationNames.h>
+#import <SentryLog.h>
+#import <SentrySDK+Private.h>
+#import <SentrySpan.h>
+#import <SentrySysctl.h>
+
+#if SENTRY_HAS_UIKIT
+
+#    import <UIKit/UIKit.h>
+
+static NSDate *appStart = nil;
+
+@interface
+SentryAppStartTracker ()
+
+@property (nonatomic, strong) SentryOptions *options;
+@property (nonatomic, strong) id<SentryCurrentDateProvider> currentDate;
+@property (nonatomic, strong) SentryAppState *previousAppState;
+@property (nonatomic, strong) SentryDispatchQueueWrapper *dispatchQueue;
+@property (nonatomic, strong) SentryAppStateManager *appStateManager;
+@property (nonatomic, strong) SentrySysctl *sysctl;
+@property (nonatomic, assign) BOOL wasInBackground;
+
+@end
+
+@implementation SentryAppStartTracker
+
++ (void)load
+{
+    // Invoked whenever this class is added to the Objective-C runtime. That's the best
+    // approximation of the app start time we can get.
+    appStart = [NSDate date];
```

Comment:
I was studying this method of getting the app start time, this could be useful.

```Objc
static CFTimeInterval processStartTime() {
    size_t len = 4;
    int mib[len];
    struct kinfo_proc kp;
    
    sysctlnametomib("kern.proc.pid", mib, &len);
    mib[3] = getpid();
    len = sizeof(kp);
    sysctl(mib, 4, &kp, &len, NULL, 0);
    
    struct timeval startTime = kp.kp_proc.p_un.__p_starttime;
    return startTime.tv_sec + startTime.tv_usec / 1e6;
}
```
To use I would do something like this:

```objc
    NSTimeInterval since1970 = [NSDate date].timeIntervalSince1970;
    CFTimeInterval starttime = processStartTime();
    long duration = since1970 - since1970;
```

Reply:
Thanks will take a look 👏

Reply:
I changed it to your sample. Thanks, @brustolin.